components/mpas-ocean/src/shared/mpas_ocn_gm.F

```diff
@@ -680,6 +682,96 @@ subroutine ocn_GM_compute_Bolus_velocity(statePool, &
          endif
 
          nEdges = nEdgesArray(3)
+         if (config_GM_horizontal_taper == 'RossbyRadius'.or. &
+             config_Redi_horizontal_taper == 'RossbyRadius') then
+            allocate (RossbyRadiusTaper(nEdges))
+            call mpas_pool_get_array(meshPool, 'fEdge', fEdge)
+
+            !$omp parallel
+            !$omp do schedule(runtime) private(k, cell1, cell2, sumN2, lt1, lt2, c_mode1, resolution)
+            do iEdge = 1, nEdges
+               cell1 = cellsOnEdge(1, iEdge)
+               cell2 = cellsOnEdge(2, iEdge)
+               sumN2 = 0.0
+
+               do k = minLevelEdgeBot(iEdge)+1, maxLevelEdgeTop(iEdge)-1
+
+                  lt1 = 0.5_RKIND*(layerThickness(k,cell1) + layerThickness(k-1,cell1))
+                  lt2 = 0.5_RKIND*(layerThickness(k,cell2) + layerThickness(k-1,cell2))
+                  sumN2 = sumN2 + 0.5_RKIND*(lt1*sqrt(max(BruntVaisalaFreqTop(k,cell1),0.0_RKIND)) + &
+                                  lt2*sqrt(max(BruntVaisalaFreqTop(k,cell2),0.0_RKIND)))
```

Comment:
@vanroekel please check on the `sqrt()` here. Note that we have
```
mpas_ocn_diagnostics.F
1163       coef = -gravity / rho_sw
1179             BruntVaisalaFreqTop(k,iCell) = coef * (displacedDensity(k-1,iCell) - density(k,iCell)) &
1180               / (zMid(k-1,iCell) - zMid(k,iCell))
```
so `BruntVaisalaFreqTop` is actually `N2 = -g/rho0 * drho/dz`. I think you really want sumN2, as implied by the summation variable, not sumN.

Reply:
You want sumN2 to be m/s for 'c_mode1' below to make sense. Just based on that, then the sqrt is correct and maybe the variable name should be changed to `sumN`.

Reply:
something still doesn't add up though, because if we remove the sqrt, then you are left with h*N2, which is m/s^2, not m^2/s^2.

Reply:
OK, I get it. We can leave this be. The summation scalar could be called `sum_hN` and then `cGMphaseSpeed` below really is the wave speed in `m/s` (I don't understand the 1/pi - is that a radians conversion). Then it is just line  721 that needs correcting.

Reply:
It looks like a radians conversion, but I don't understand where it came in. Maybe f and beta have radians in them?

Reply:
f is radians/sec. I added lengthy comments to the code on the source of these equations, all below, and refer to both Chelton (1998) and Hallberg (2013) with equation numbers.